### ISSUE TITLE
Look for config in package.json

### ISSUE THREAD

**Issue by max** [AtomLinter/linter-jshint#25]
JSHint supports a [jshintConfig](http://www.jshint.com/blog/better-npm-integration/) key in package.json but linter-jshint doesn't. I tried to figure out how to get this to work but because the linted file seems to be in a tmp directory without the package.json file JSHint won't automatically pick it up. Unfortunately you can't pass that file in the form of a flag.

Does anyone have suggestions on how to add this functionality?

**Comment by harrytruong** [comment]
@iam4x I assume you didn't get a chance to look into this issue, so I came up with a fix.

@max I believe you had the right solution. Based on https://github.com/jshint/jshint/pull/1503, adding the `--filename` argument and passing the file via `stdin` is the proper fix. 

Unfortunately, `linter-jshint` doesn't use `stdin`. In fact, `linter` doesn't allow for `stdin` at all. (`lintFile` in `Linter` will spawn the process, but doesn't allow us hook into `stdin`.)

I submitted a simple PR to the `AtomLinter/Linter` repo:

> **https://github.com/AtomLinter/Linter/pull/556:**
> **add public method "afterSpawnProcess" to Linter class**
>
> Gives subclasses a chance to read or change the child
> process, after creating new BufferedProcess while lintFile.
> 
> Useful for passing data via stdin (process.process.stdin).
> 
> This is a backward compatible improvement.

Assuming that PR is merged, I submitted the proper PR fix here: 
> **[AtomLinter/linter-jshint#97](https://github.com/AtomLinter/linter-jshint/pull/97)**
>**call jshint with stdin, instead of filePath arg**

**Comment by harrytruong** [comment]
@steelbrain has pointed out that this issue was fixed, as of the latest rewrite. 

I've confirmed that `jshint` now properly loads `jshintConfig` from within `package.json`:

```js
// package.json
{
    // ...
    "jshintConfig": {
        "esnext": true,
        "undef": true,
        "unused": true,
        "predef": [ "MY_GLOBAL" ]
        // etc...
    }
}
```

This issue can be closed now.

**Comment by max** [comment]
Thanks everyone! :fireworks: :facepunch:
